sematic/ui/src/pipelines/PipelineMetricsPanel.tsx

```diff
@@ -0,0 +1,143 @@
+import { Box, Typography } from "@mui/material";
+import { useCallback, useContext, useEffect, useMemo, useState } from "react";
+import { UserContext } from "src/appContext";
+import Loading from "src/components/Loading";
+import { ExtractContextType } from "src/components/utils/typings";
+import { Run } from "src/Models";
+import { CompactMetrics, CompactMetricsPayload } from "src/Payloads";
+import { fetchJSON } from "src/utils";
+import { usePipelineRunContext } from "../hooks/pipelineHooks";
+import PipelineRunViewContext from "./PipelineRunViewContext";
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend,
+  ChartData
+} from 'chart.js';
+import { Line } from 'react-chartjs-2';
+import { metricsSocket } from "src/sockets";
+
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend
+);
+
+const COLORS = [
+  'rgb(255, 99, 132)',
+  'rgb(54, 162, 235)',
+  'rgb(255, 206, 86)',
+  'rgb(75, 192, 192)',
+  'rgb(153, 102, 255)',
+  'rgb(255, 159, 64)'
+];
+
+export default function PipelineMetricsPanel() {
+  const { rootRun } = usePipelineRunContext () as ExtractContextType<typeof PipelineRunViewContext > & {
+    rootRun: Run
+  };
+
+  const { user } = useContext(UserContext);
+  const [metrics, setMetrics] = useState<CompactMetrics | undefined>(undefined);
+
+  const refreshMetrics = useCallback(() => {
+    if (rootRun === undefined) return;
+    fetchJSON({
```

Comment:
1. Can this be changed to `useHttpClient()` ?
2. You can package this into a hook, with the use `useAsyncRetry` can help reduce a lot of boilerplate code. Using a hook also has a better separation of behavioral and presentation code. The logic in this file is very similar to `graphHook.ts`, maybe take a look at that.   
3. Ditto for `RunMetricsPanel.tsx`